lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntime.cpp

```diff
@@ -608,6 +613,51 @@ bool SwiftLanguageRuntimeImpl::AddJitObjectFileToReflectionContext(
       });
 }
 
+bool SwiftLanguageRuntimeImpl::AddObjectFileToReflectionContext(
+    ModuleSP module, ObjectFile &obj_file, llvm::Triple::ObjectFormatType obj_format_type) {
+  auto obj_file_format = GetObjectFileFormat(obj_format_type);
+
+  auto &memory_reader = m_reflection_ctx->getReader();
+  auto *lldb_memory_reader = (LLDBMemoryReader *)&memory_reader; 
+  auto start_address = lldb_memory_reader->addModuleToAddressMap(module);
+
+  auto *section_list = obj_file.GetSectionList();
+  auto segment_iter = llvm::find_if(*section_list, [](auto segment) {
+    return segment->GetName() == "__TEXT";
+  });
+
+  if (segment_iter == section_list->end())
+    return false;
+
+  auto *segment = segment_iter->get();
+
+  return m_reflection_ctx->addImage(
+      [&](swift::ReflectionSectionKind section_kind)
+          -> std::pair<swift::remote::RemoteRef<void>, uint64_t> {
+        auto section_name =
+            ConstString(obj_file_format->getSectionName(section_kind));
```

Comment:
why is the ConstString necessary here?

Reply:
Because there's no ConstString overload that compares a ConstString to a StringRef.

Reply:
It would be more efficient to use `if (section->GetName().AsStringRef() == section_name)` than looking up and potentially allocating a new ConstString.